bundle/src/main/java/com/adobe/acs/commons/i18n/impl/I18nProviderImpl.java

```diff
@@ -0,0 +1,207 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.i18n.impl;
+
+
+import com.adobe.acs.commons.i18n.I18nProvider;
+import com.adobe.acs.commons.models.injectors.impl.InjectorUtils;
+import com.adobe.acs.commons.util.impl.AbstractGuavaCacheMBean;
+import com.adobe.acs.commons.util.impl.CacheMBean;
+import com.adobe.acs.commons.util.impl.exception.CacheMBeanException;
+import com.day.cq.i18n.I18n;
+import com.day.cq.wcm.api.Page;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.commons.osgi.Order;
+import org.apache.sling.commons.osgi.PropertiesUtil;
+import org.apache.sling.commons.osgi.RankedServices;
+import org.apache.sling.i18n.ResourceBundleProvider;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.metatype.annotations.Designate;
+
+import javax.management.DynamicMBean;
+import javax.management.NotCompliantMBeanException;
+import javax.management.openmbean.CompositeType;
+import javax.management.openmbean.OpenDataException;
+import javax.management.openmbean.OpenType;
+import javax.management.openmbean.SimpleType;
+import javax.servlet.http.HttpServletRequest;
+import java.util.Locale;
+import java.util.Map;
+import java.util.ResourceBundle;
+import java.util.concurrent.TimeUnit;
+
+@Component(
+        property= {
+            "jmx.objectname=com.adobe.acs.httpcache:type=I18N Provider Cache"
+        },
+        service = {I18nProvider.class,DynamicMBean.class},
+        immediate = true,
+        reference = {
+            @Reference(
+                    name="resourceBundleProviders",
+                    service = ResourceBundleProvider.class,
+                    cardinality = ReferenceCardinality.AT_LEAST_ONE,
+                    policy = ReferencePolicy.DYNAMIC,
+                    bind="bindResourceBundleProvider",
+                    unbind="unbindResourceBundleProvider"
+            )
+        }
+)
+@Designate(ocd = Config.class)
+public class I18nProviderImpl extends AbstractGuavaCacheMBean<String,I18n> implements I18nProvider, DynamicMBean {
+
+    private static final String JMX_PN_I18N = "I18n Object";
+
+    private final RankedServices<ResourceBundleProvider> resourceBundleProviders = new RankedServices<>(Order.ASCENDING);
+
+    private Cache<String, I18n> cache;
+
+    public I18nProviderImpl() throws NotCompliantMBeanException {
+        super(CacheMBean.class);
+    }
+
+    protected void activate(Map<String, Object> properties)
```

Comment:
Why do you use the @ocd annotation but then you fall back to the "old" signature?

Reply:
good question.. the default values don't work yet for some weird reason. specified values do however.

see: https://github.com/Adobe-Consulting-Services/acs-aem-commons/issues/1652

Reply:
is this still the case?

Reply:
Not anymore :) I guess I'll change this now.

Reply:
Any chance we can up the test coverage just a little to make the checks happy?

Reply:
Please cut me some slack :) I improve the test coverage in the next merge requests, this merge request is just a "slice" PR so it doesn't become to big at one time.

Plus I added a unit test on another PR as well.

Reply:
No worries.  Actually if you merge the main branch into your feature branch it might break even.

Reply:
Ok done. Actually the code coverage should increase, but 1 test is powermocked. Is it picked up?